content/4_Gold/Intro_DP.mdx

```diff
@@ -132,11 +128,46 @@ It's usually a good idea to write a slower solution first. For example, if the c
 
 </Info>
 
-<IncompleteSection>
+**Dynamic Programming** (DP) is an important algorithmic technique in Competitive Programming from the gold division to competitions like the International Olympiad of Informatics. By breaking down the full task into sub-problems, DP avoids the redundant computations of brute force solutions.
+
+Although it is not too difficult to grasp the general ideas behind DP, the technique can be used in a diverse range of problems and is a must-know idea for competitors in the USACO Gold division.
+
+Often, dynamic programming can be split into 3 steps: the base cases, the recurrence, and the answer. 
+
+## Base Cases
+
+The base cases are the cases of our dynamic programming that cannot be defined in terms of other cases of the same function. Typically, we set these cases manually before performing our recurrence in order to ensure that our recurrence works properly.
+
+## Recurrence/Transition
+
+The recurrence of our dynamic programming is the way we define each case in the terms of other cases of the same function. We always want the recurrence to be made up of cases that we have computed before our current case. 
+
+## The Answer
+
+Lastly, we want to figure out which case of our dynamic programming is the answer to our problem. This is usually an intuitive step of dynamic programming and will almost always come naturally. 
+
+## Solution - The $N$th Fibonacci Term Problem
+
+A common introductory problem for dynamic programming is finding the N-th Fibonacci term. This can be done with recursion with the formula $F_n$ = $F_{n-1}$ + $F_{n-2}$ in $O(N^2)$ by simply recursing down to 0 for each term. However, notice that we are repeating some steps for each term. In order to reduce our complexity, we can use dynamic programming. 
```

Comment:
```suggestion
A common introductory problem for dynamic programming is finding the N-th Fibonacci term. This can be done with recursion with the formula $F_n = F_{n-1} + F_{n-2}$ in $O(N^2)$ by simply recursing down to 0 for each term. However, notice that we are repeating some steps for each term. In order to reduce our complexity, we can use dynamic programming. 
```